runelite-client/src/main/java/net/runelite/client/plugins/lmsperformancetracker/FightPerformance.java

```diff
@@ -0,0 +1,187 @@
+	}
+
+	// used in getTestInstance, used for debugging purposes.
+	private FightPerformance(String playerName, String opponentName, int playerAttackCount, int playerSuccessCount, int opponentAttackCount, int opponentSuccessCount)
+	{
+		this.playerName = playerName;
+		this.opponentName = opponentName;
+		this.playerAttackCount = playerAttackCount;
+		this.playerSuccessCount = playerSuccessCount;
+		this.playerSuccessRate = (double) this.playerSuccessCount / (double) this.playerAttackCount * 100.0;
+		this.playerDied = false;
+		this.opponentAttackCount = opponentAttackCount;
+		this.opponentSuccessCount = opponentSuccessCount;
+		this.opponentSuccessRate = (double) this.opponentSuccessCount / (double) this.opponentAttackCount * 100.0;
+		this.opponentDied = false;
+		this.timeFightEnded = Instant.now();
+	}
+
+	public void addAttack(String playerName, boolean success)
+	{
+		if (playerName.equals(this.playerName))
+		{
+			playerAttackCount++;
+			if (success)
+			{
+				playerSuccessCount++;
+			}
+			playerSuccessRate = (double) playerSuccessCount / (double) playerAttackCount * 100.0;
+		}
+		else if (playerName.equals(opponentName))
+		{
+			opponentAttackCount++;
+			if (success)
+			{
+				opponentSuccessCount++;
+			}
+			opponentSuccessRate = (double) opponentSuccessCount / (double) opponentAttackCount * 100.0;
+		}
+	}
+
+	public void playerDied()
+	{
+		playerDied = true;
+	}
+
+	public void opponentDied()
+	{
+		opponentDied = true;
+	}
+
+	public void fightEnded()
+	{
+		timeFightEnded = Instant.now();
+	}
+
+	// Return a simple string to display the current player's success rate.
+	// ex. "42/59 (71%)". The name is not included as it will be in a separate view.
+	// Would round to 1 decimal space, but the overlay size is restrictive.
+	public String getPlayerStatsString()
+	{
+		// The success rate is the percentage of successful attacks.
+		return playerSuccessCount + "/" + playerAttackCount + " (" + Math.round(playerSuccessRate) + "%)";
+	}
+
+	// Return a simple string to display the current opponent's success rate.
+	public String getOpponentStatsString()
+	{
+		// The success rate is the percentage of successful attacks.
+		return opponentSuccessCount + "/" + opponentAttackCount + " (" + Math.round(opponentSuccessRate) + "%)";
+	}
+
+	// returns true if player success rate > opponent success rate.
+	// could be "wrong" in some cases, if someone is eating a lot and not actually attacking much,
+	// they could have a higher success rate than the person clearly winning. Although it is hard to
+	// judge by comparing attack counts since someone using rune knives/msb against ballista/ags
+	// is not uncommon, which could cause a situation opposite of the one described above.
+	public boolean playerWinning()
+	{
+		return playerSuccessRate > opponentSuccessRate;
+	}
+
+	// returns true if opponent success rate > player success rate.
+	public boolean opponentWinning()
+	{
+		return opponentSuccessRate > playerSuccessRate;
+	}
```

Comment:
This was intentional, since I wanted to remove the green highlight it's a tie. If I did it that way and it was a tie, the opponent would be highlighted as winning.